internal/controller/controllers/agent_controller.go

```diff
@@ -213,6 +213,11 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 		return r.updateStatus(ctx, log, agent, origAgent, &h.Host, h.ClusterID, err, true)
 	}
 
+	err = r.updateClusterDeploymentRefLabels(log, ctx, agent)
```

Comment:
I would move the call to updateInventoryLabels() from inside updateInventory() to here, and then unite updateInventoryLabels() and updateClusterDeploymentRefLabels() to just be updateLabels().  I think that makes the code more organized and we also update the resource only once.

Reply:
Sure thing, just refactored it and put it all into a single function call